src/EditorFeatures/CSharpTest/AssignOutParameters/AssignOutParametersTests.cs

```diff
@@ -0,0 +1,534 @@
+    {
+        internal override (DiagnosticAnalyzer, CodeFixProvider) CreateDiagnosticProviderAndFixer(Workspace workspace)
+            => (null, new AssignOutParametersCodeFixProvider());
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForSimpleReturn()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        [|return 'a';|]
+    }
+}",
+@"class C
+{
+    char M(out int i)
+    {
+        i = 0;
+        return 'a';
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestForSwitchSectionReturn()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        switch (0)
+        {
+            default:
+                [|return 'a';|]
+        }
+    }
+}",
+@"class C
+{
+    char M(out int i)
+    {
+        switch (0)
+        {
+            default:
+                i = 0;
+                return 'a';
+        }
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestMissingWhenVariableAssigned()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        i = 0;
+        [|return 'a';|]
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestWhenNotAssignedThroughAllPaths1()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    char M(bool b, out int i)
+    {
+        if (b)
+            i = 1;
+        
+        [|return 'a';|]
+    }
+}",
+@"class C
+{
+    char M(bool b, out int i)
+    {
+        if (b)
+            i = 1;
+        i = 0;
+        return 'a';
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestWhenNotAssignedThroughAllPaths2()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    bool M(out int i1, out int i2)
+    {
+        [|return Try(out i1) || Try(out i2);|]
+    }
+
+    bool Try(out int i)
+    {
+        i = 0;
+        return true;
+    }
+}",
+@"class C
+{
+    bool M(out int i1, out int i2)
+    {
+        i2 = 0;
+        return Try(out i1) || Try(out i2);
+    }
+
+    bool Try(out int i)
+    {
+        i = 0;
+        return true;
+    }
+}");
+        }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsAssignOutParameters)]
+        public async Task TestMissingWhenAssignedThroughAllPaths()
+        {
+            await TestMissingInRegularAndScriptAsync(
+@"class C
+{
+    char M(out int i)
+    {
+        if (b)
+            i = 1;
+        else
+            i = 2;
+        
+        [|return 'a';|]
```

Comment:
Can we also have a test where the out param is passed as an out argument? Say:

```csharp
        if (b)
            i = 1;
        else
            SomeMethod(out i);
```

Reply:
added.